projects/plugins/automattic-for-agencies-client/src/js/components/connection-button/index.jsx

```diff
@@ -0,0 +1,33 @@
+import { TermsOfService } from '@automattic/jetpack-components';
+import { ConnectButton } from '@automattic/jetpack-connection';
+import { __ } from '@wordpress/i18n';
+import React from 'react';
+import styles from './styles.module.scss';
+
+/**
+ * Connection Button component.
+ * Displays a button to connect the site, along with the Terms of Service blurb.
+ *
+ * @param {object} props           - Component props
+ * @param {boolean} props.hideTOS  - Flag to hide the Terms of Service text.
+ * @returns {React.Component} The `ConnectionButton` component.
+ */
+export default function ConnectionButton( { hideTOS = false } ) {
+	return (
+		<div className={ styles[ 'site-connection' ] }>
+			{ ! hideTOS && (
+				<div className={ styles[ 'terms-of-service-wrapper' ] }>
+					<TermsOfService
+						agreeButtonLabel={ __( 'connect this site', 'automattic-for-agencies-client' ) }
+						className={ styles[ 'terms-of-service' ] }
+					/>
```

Comment:
In a follow-up PR, we need to replace this button or create a new one that behaves differently when clicking on `share details` link. Currently, the link leads users to a page outside of the plugin, but it should keep them in the same place and show them what pieces of information we will sync (see the Figma file linked in the issue).